polaris-react/src/components/Box/Box.tsx

```diff
@@ -0,0 +1,263 @@
+import React, {ReactNode, forwardRef} from 'react';
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
+import type {depth, shape, spacing} from '@shopify/polaris-tokens';
+
+import {classNames} from '../../utilities/css';
+
+import styles from './Box.scss';
+
+type Background =
+  | 'background'
+  | 'backgroundHovered'
+  | 'backgroundPressed'
+  | 'backgroundSelected'
+  | 'surface'
+  | 'surfaceDark'
+  | 'surfaceNeutral'
+  | 'surfaceNeutralHovered'
+  | 'surfaceNeutralPressed'
+  | 'surfaceNeutralDisabled'
+  | 'surfaceNeutralSubdued'
+  | 'surfaceNeutralSubduedDark'
+  | 'surfaceSubdued'
+  | 'surfaceDisabled'
+  | 'surfaceHovered'
+  | 'surfaceHoveredDark'
+  | 'surfacePressed'
+  | 'surfacePressedDark'
+  | 'surfaceDepressed'
+  | 'surfaceSearchField'
+  | 'surfaceSearchFieldDark'
+  | 'backdrop'
+  | 'overlay';
+
+type DepthTokenGroup = typeof depth;
+type DepthTokenName = keyof DepthTokenGroup;
+type ShadowsTokenName = Exclude<DepthTokenName, `shadows-${string}`>;
+
+type DepthTokenScale = ShadowsTokenName extends `shadow-${infer Scale}`
+  ? Scale
+  : never;
+
+type ShapeTokenGroup = typeof shape;
+type ShapeTokenName = keyof ShapeTokenGroup;
+
+type BorderShapeTokenScale = ShapeTokenName extends `border-${infer Scale}`
+  ? Scale
+  : never;
+
+type BorderTokenScale = Exclude<
+  BorderShapeTokenScale,
+  `radius-${string}` | `width-${string}`
+>;
+
+interface Border {
+  bottom: BorderTokenScale;
+  left: BorderTokenScale;
+  right: BorderTokenScale;
+  top: BorderTokenScale;
+}
+
+type BorderRadiusTokenScale = Extract<
+  BorderShapeTokenScale,
+  `radius-${string}`
```

Comment:
Spacing takes a numerical string without prefix but border radius requires one:
```ts
padding="1"
borderRadius="radius-4"
```

Can we make it consistent? I'm fine with `space-1` or `borderRadius="4"` Whatever you think makes the most sense.

Reply:
What is `base`? I always disliked `base` as a value...

Reply:
Right LOL. I think that is a token naming convention we should challenge as it's not clear to the user what it does.

Reply:
100%, I know we had previous discussions about modifying some of the naming structure for tokens parallel to that work so that can be included there as well 👍